### PR TITLE
feat(connection-form): allow editing connected connections, but only the name, colour and favorite checkbox COMPASS-8160

### PR DESCRIPTION
**Pull request by lerouxb**
See [figma design](https://www.figma.com/proto/JmNVeRAx398lbu3kNQiuZc/Multiple-connection-UX-tweaks?page-id=0%3A1&node-id=5-1995&node-type=canvas&viewport=25%2C421%2C0.62&t=FN4NntbaFlOMRK7Q-1&scaling=min-zoom&content-scaling=fixed&starting-point-node-id=5%3A1995&show-proto-sidebar=1) and [slack thread](https://mongodb.slack.com/archives/C05MLSFMUJC/p1728388841356229).

Screenshot 2024-10-09 at 12 33 01

### CODE REVIEW

File: packages/connection-form/src/components/connection-form-actions.tsx [hunk 1]
```diff
@@ -89,7 +89,11 @@ export function ConnectionFormModalActions({
           <div className={saveAndConnectStyles}>
             <Button
               data-testid="save-button"
-              variant={ButtonVariant.PrimaryOutline}
+              variant={
+                onSaveAndConnect
```
Comment (lerouxb) [inline packages/connection-form/src/components/connection-form-actions.tsx:93]: Basically: If there is no save&connect button (the Connect button), make the Save button the "default" (ie. solid green) one. type for both are still implicitly "button", though. there is no default. Would be ambiguous anyway because we have a custom onSubmit handler that would have to be kept in sync.

File: packages/connection-form/src/components/connection-form.tsx [hunk 2]
```diff
@@ -429,16 +422,58 @@ function ConnectionForm({
...
   );
 
   const showHelpCardsInForm = useConnectionFormSetting('showHelpCardsInForm');
 
+  /*
+  To test onSubmit on a form (ie. what happens when you press {enter} and not
+  what happens when you click a submit button) you have to use submit() which only
+  works with role="form", but eslint doesn't like that because role="form"
+  should be redundant.
+  https://stackoverflow.com/questions/59362804/pressing-enter-to-submit-form-in-react-testing-library-does-not-work
```
Comment (lerouxb) [inline packages/connection-form/src/components/connection-form.tsx:470]: If there's a better way to simulate submitting a form by pressing enter rather than pressing a button so that the callback gets called in tests, please let me know.

File: packages/connection-form/src/components/connection-form.tsx [hunk 3]
```diff
@@ -451,7 +486,8 @@ function ConnectionForm({
         <div className={formHeaderStyles}>
           <div className={formHeaderContentStyles}>
             <H3 className={headingWithHiddenButtonStyles}>
-              {initialConnectionInfo.favorite?.name ?? 'New Connection'}
+              {(initialConnectionInfo.favorite?.name ?? 'New Connection') ||
+                'Edit Connection'}
```
Comment (lerouxb) [inline packages/connection-form/src/components/connection-form.tsx:490]: Drive-by to fix the case where you edit a connection that has no favorite name.

Screenshot 2024-10-08 at 14 14 53

File: packages/connection-form/src/components/form-help/form-help.tsx [hunk 4]
```diff
@@ -29,13 +30,13 @@ const descriptionStyles = css({
 });
 
 const sectionDarkModeStyles = css({
-  '--theme-background-color': palette.blue.dark3,
-  '--theme-border-color': palette.blue.dark2,
+  '--theme-background-color': palette.gray.dark3,
```
Comment (lerouxb) [inline packages/connection-form/src/components/form-help/form-help.tsx:33]: Note that according to the design, the blue help text boxes on the right are now grey.

File: packages/connection-form/src/components/connection-form.tsx [hunk 5]
```diff
@@ -303,10 +315,12 @@ type ConnectionFormPropsWithoutSettings = {
   darkMode?: boolean;
   initialConnectionInfo: ConnectionInfo;
   connectionErrorMessage?: string | null;
+  disableEditingConnectedConnection?: boolean;
```
Comment (lerouxb) [inline packages/connection-form/src/components/connection-form.tsx:318]: Optional new prop with undefined acting like false, so vscode should be unaffected.

File: packages/connection-form/src/components/connection-form.tsx [hunk 6]
```diff
@@ -461,9 +497,36 @@ function ConnectionForm({
...
         <div className={formContentContainerStyles}>
           <div className={formContentStyles}>
+            {disableEditingConnectedConnection && onDisconnectClicked && (
+              <Banner
+                data-testid="disabled-connected-connection-banner"
+                className={disabledConnectedConnectionBannerStyles}
+              >
+                <div className={disabledConnectedConnectionContentStyles}>
+                  <div>
+                    While connected, you may only personalize your
```
Comment (addaleax) [inline packages/connection-form/src/components/connection-form.tsx:507]: This isn't really what "personalize" means, unless we consider connections to be people

**Review by addaleax — APPROVED** [review]

File: packages/connection-form/src/components/connection-form.tsx (hunk 6 above)
Comment (lerouxb) [reply packages/connection-form/src/components/connection-form.tsx:507]: Fits the dictionary definition.
